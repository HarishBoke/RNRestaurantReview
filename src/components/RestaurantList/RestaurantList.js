import React from 'react';
import {View, Text, ScrollView} from 'react-native';
import RestaurantListStyle from './RestaurantList.style';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';

const RestaurantRow = (place, index) => (
  <View key={place.get('name')} style={RestaurantListStyle.sectionContainer}>
    <View style={RestaurantListStyle.flexColumn}>
      <Text style={RestaurantListStyle.sectionCount}>{`${index + 1}. `}</Text>
      <Text style={RestaurantListStyle.sectionTitle}>{place.get('name')}</Text>
      <Text style={RestaurantListStyle.sectionButton}>Info</Text>
    </View>
    <View>
      <Text style={RestaurantListStyle.sectionSubTitle}>
        {place.get('address')}
      </Text>
    </View>
  </View>
);
// list: use react native list
const RestaurantList = (props) => {
  const {restaurants, restaurantSearch} = props;
  return (
    <ScrollView
      contentInsetAdjustmentBehavior="automatic"
      style={RestaurantListStyle.scrollView}>
      <View style={RestaurantListStyle.body}>
        {restaurants
          .filter((place) => {
            return (
              !restaurantSearch ||
              place
                .get('name')
                .toLowerCase()
                .indexOf(restaurantSearch.toLowerCase()) > -1
            );
          })
          .map((place, index) => RestaurantRow(place, index))}
      </View>
    </ScrollView>
  );
};
RestaurantList.propTypes = {
  restaurants: ImmutablePropTypes.list,
  restaurantSearch: PropTypes.string,
};

export default RestaurantList;
