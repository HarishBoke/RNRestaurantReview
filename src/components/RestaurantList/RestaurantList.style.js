import {Colors} from 'react-native/Libraries/NewAppScreen';

export default {
  scrollView: {
    backgroundColor: Colors.lighter,
  },
  sectionContainer: {
    paddingHorizontal: 20,
    paddingVertical: 10,
    borderColor: '#ddd',
    borderBottomWidth: 1,
  },
  sectionCount: {
    fontSize: 18,
    fontWeight: '800',
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: '600',
    color: Colors.black,
    flex: 6,
  },
  sectionButton: {
    fontWeight: '600',
    borderWidth: 1,
    borderColor: '#0066cc',
    borderRadius: 18,
    paddingHorizontal: 10,
    paddingVertical: 3,
    backgroundColor: '#fff',
    textAlign: 'center',
  },
  sectionButtonText: {
    color: '#0066cc',
    fontSize: 12,
  },
  sectionSubTitle: {
    fontSize: 16,
    fontWeight: '400',
    color: '#999',
    paddingLeft: 20,
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
    color: Colors.dark,
  },
  highlight: {
    fontWeight: '700',
  },
  footer: {
    color: Colors.dark,
    fontSize: 12,
    fontWeight: '600',
    padding: 4,
    paddingRight: 12,
    textAlign: 'right',
  },
  flexColumn: {
    flexDirection: 'row',
  },
  infoMessage: {
    textAlign: 'center',
    borderWidth: 1,
    borderRadius: 4,
    borderColor: '#ddd',
    marginHorizontal: 40,
    marginVertical: 10,
    padding: 10,
    backgroundColor: '#fff',
  },
};
