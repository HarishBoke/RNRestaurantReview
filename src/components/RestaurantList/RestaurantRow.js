import React, {useState} from 'react';
import {View, Text, Button} from 'react-native';
import RestaurantListStyle from './RestaurantList.style';

const RestaurantRow = (props) => {
  const {place, index} = props;
  const [showInfo, setShowInfo] = useState(false);
  return (
    <React.Fragment>
      {showInfo && (
        <Text style={RestaurantListStyle.infoMessage}>Show info</Text>
      )}
      <View key={place.name} style={RestaurantListStyle.sectionContainer}>
        <View style={RestaurantListStyle.flexColumn}>
          <Text style={RestaurantListStyle.sectionCount}>{`${
            index + 1
          }. `}</Text>
          <Text style={RestaurantListStyle.sectionTitle}>{place.name}</Text>
          <Button
            style={RestaurantListStyle.sectionButton}
            title="Info"
            accessibilityLabel="Info"
            onPress={() => setShowInfo(!showInfo)}
          />
        </View>
        <View>
          <Text style={RestaurantListStyle.sectionSubTitle}>
            {place.address}
          </Text>
        </View>
      </View>
    </React.Fragment>
  );
};

export default RestaurantRow;
