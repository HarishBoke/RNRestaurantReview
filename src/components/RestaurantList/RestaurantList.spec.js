import React from 'react';
import {shallow} from 'enzyme';
import Immutable from 'immutable';

import RestaurantList from './RestaurantList';

const restaurants = Immutable.fromJS([
  {name: 'React Cafe', address: '123, Main street'},
  {name: 'React Native Garden', address: '098, North Main street'},
]);
const restaurantSearch = 'React Cafe';

const initApp = (overrides) => {
  const mockProps = {restaurants, restaurantSearch};
  const mockMethods = {};

  const wrapper = shallow(
    <RestaurantList {...mockMethods} {...mockProps} {...overrides} />,
  );

  return {mockMethods, mockProps, wrapper};
};

describe('Component: RestaurantList', () => {
  describe('rendering RestaurantList', () => {
    it('Snapshot: should display the RestaurantList', async () => {
      const {wrapper} = initApp();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
