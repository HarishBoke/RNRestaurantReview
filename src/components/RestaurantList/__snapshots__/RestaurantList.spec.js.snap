// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RestaurantList rendering RestaurantList Snapshot: should display the RestaurantList 1`] = `ShallowWrapper {}`;
